### ISSUE TITLE
RTDB emulator doesn't work properly with databaseAuthVariableOverride

### ISSUE THREAD

**Issue by rhodgkins** [firebase/firebase-tools#2554]
### [REQUIRED] Environment info

**firebase-tools:** 8.7.0
**Platform:** macOS
**node:** v10.22.0
**firebase-admin:** 9.1.0

### [REQUIRED] Test case

Add some rules like the following:

```json
{
   "rules": {
      "$userId": {
         // Must have logged in in the last hour and accessing own data, or have claims to access the data or have public access
         ".read": "(auth.token.auth_time >= ((now / 1000) - 60 * 60) && ($userId === auth.uid || auth.token.customClaimAccess === $userId)) || data.child('publicAccess').val() === true"
      }
   }
}
```

### [REQUIRED] Steps to reproduce

```js
const admin = require('firebase-admin')

const noAuthApp = admin.initializeApp({
   databaseURL: 'https://blah.firebaseio.com',
   databaseAuthVariableOverride: null,
}, 'no-auth')
const userAuthApp = admin.initializeApp({
   databaseURL: 'https://blah.firebaseio.com',
   databaseAuthVariableOverride: {
      uid: 'USER_ID',
   },
}, 'user-auth')
const customClaimsAuthApp = admin.initializeApp({
   databaseURL: 'https://blah.firebaseio.com',
   databaseAuthVariableOverride: {
      token: {
         customClaimAccess: 'USER_ID',
      },
   },
}, 'custom-claims-auth')

Promise.all([
   noAuthApp.database().ref('USER_ID').once('value'),
   userAuthApp.database().ref('USER_ID').once('value'),
   customClaimsAuthApp.database().ref('USER_ID').once('value'),
]).then(results => {
   console.log(results)
}, error => {
   console.error(error)
}).finally(() => Promise.all(admin.apps.map(a => a.delete())))
```

1. Run emulator `firebase --project blah emulators:start --only database` (can't use `emulators:exec` as need to keep it running to view the rule evaluations).
2. Run the above script against the emulator `FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000 node script.js`.
3. View the evaluated DB rules: `http://localhost:9000/.inspect/coverage?ns=blah`

### [REQUIRED] Expected behavior

I'd expect the latter rule (`data.child('publicAccess').val() === true`) to be evaluated for all 3 fetches - can by hovering over the last rule: ![image](https://user-images.githubusercontent.com/1810588/90509449-3e2da200-e151-11ea-9c42-740caa57eb23.png).

### [REQUIRED] Actual behavior

`once('value')` is rejected with `Error: permission_denied at /USER_ID: Client doesn't have permission to access the desired data.`

Note that adding in the extra [`auth` info](https://firebase.google.com/docs/reference/security/database#auth) to the `databaseAuthVariableOverride` makes everything work but this should not be needed (and how does that work when `null` is used for no auth)?

**Comment by rhodgkins** [comment]
Just to add to this, I think the missing evaluations might be to do with `null` values because if you have a rule such as `data.child('blah').val() < 100 || data.child('other').exists()` and `blah` is unset (`null`) the hover info will show `.val()` returning `null` each time but the result count of `data.child('blah').val() < 100` won't match this count.

**Comment by akauppi** [comment]
I'm also tackling with similar.

If you're still on this (2 years later!!!!):

- are you sure the calls lead to the emulators?  `.firebaseio.com` looks suspicious 
- By using `demo-blah` (anything starting with `demo-`) one makes sure the calls will not aim at the real Firebase project in the cloud.

**Comment by rhodgkins** [comment]
@akauppi I don't understand your points sorry. The `.firebaseio.com` hostname is swapped out by the internals of the SDK when `FIREBASE_DATABASE_EMULATOR_HOST` envar is specified. I imagine its something to do with `null` checks in the emulator etc. etc. that doesn't quite match production.

**Comment by maneesht** [comment]
@rhodgkins - have you tried running this with the latest version of firebase-admin and firebase-database-emulator?

**Comment by rhodgkins** [comment]
@maneesht yep, if you run the reproduction steps above the same thing happens.

Ran `firebase setup:emulators:database` to get the latest DB emulator (it downloaded `v4.8.0`).

**Comment by maneesht** [comment]
@rhodgkins thank you for your patience. We might have identified what the problem is, however we do not have the bandwidth to support this right now. I will keep this open until we can look further into this.
